### PR TITLE
Vote System Rework - Remove Observer/New Player Auto-Transfer Votes

### PR DESCRIPTION
**Pull request by Maldaris**
## About The Pull Request

Adds a `observer_vote_allowed` parameter to `initiate_vote` so that (specifically) players that are either observing or in the lobby are not allowed to participate in the automatic vote to transfer.

## Why It's Good For The Game

People who are participating in the round are the only ones (besides admins) who should have a say in calling the shuttle.

## Changelog
:cl:
fix: Observers and Lobby Players cannot vote for auto transfer votes.
tweak: Auto-Transfer votes won't trigger if a transfer is already in progress.
/:cl:

### CODE REVIEW

**Comment by TheNeoGamer42** [comment]
speedmerge

**Comment by AsciiSquid** [comment]
speedmerge

**Comment by rabbetbunzel** [comment]
yes

**Comment by ADepressedPyroshark** [comment]
No funny text to insult you for observing or still being in the lobby 0/10 (please correct me if wrong)

**Comment by Maldaris** [comment]
@dee-dubya better yet, you are completely oblivious to the vote even existing.

**Comment by ADepressedPyroshark** [comment]
@Maldaris Will the noise that plays when a vote starts still play?

**Comment by Maldaris** [comment]
Good question I'll take a look at that.

**Comment by dinnerwoerror** [comment]
ah nah mate, i get it, the joke with this pull request is that the crew transfer and lone ops were meant to prevent endless five hour greenshifts where nothing happened and the round was never ended because nothing was happening, but you see, i read the wikipedia article for irony and i know that this is irony https://en.wikipedia.org/wiki/Irony because irony is a rhetorical device in which what on the surface appears to be the case or to be expected differs radically from what is actually the case, and this particular case here is situational irony, in within lies a sharp discrepancy between the expected result and actual results in a certain situation, leading to a poignantly contrary result to what is expected, yet is what is inherently expected from an instance such as this under the teachings of søren kierkegaard who proposed that this situation of irony shown here is to expose itself as the infinite absolute negativity, it is pure negativity, because it only negates; it is infinite, because it does not negate this or that phenomenon; it is absolute, because that by virtue of which it negates is a higher something that still is not, and the irony here established nothing, because that which is to be established lies behind it...

the irony shown here must "swallow its own stomach", for lack of a better phrase, as irony entails endless reflection and violent reversals, and ensures incomprehensibility at the moment it compels any speech...

**Review by MarkSuckerberg — APPROVED** [review]
Seems legit

**Comment by Maldaris** [comment]
> Seems legit
You're mostly correct, except I'm an idiot and can drop that `ghost_vote_default_ok` because there's a parameter to the function that does the same thing 😬

**Comment by SynnGraffkin** [comment]
I haven't really paid attention to it, but does this effect observers being able to vote on map votes at the end of a round?

**Comment by MarkSuckerberg** [comment]
I know the feel, good thing you noticed it before I hit the big green button

File: code/controllers/subsystem/vote.dm [hunk 1]
```diff
@@ -254,18 +263,42 @@ SUBSYSTEM_DEF(vote)
...
-			var/client/C = c
-			var/datum/action/vote/V = new
-			if(question)
-				V.name = "Vote: [question]"
-			C.player_details.player_actions += V
-			V.Grant(C.mob)
-			generated_actions += V
-		return TRUE
+		var/vote_message =  "\n<font color='purple'><b>[text]</b>\nType <b>vote</b> or click <a href='?src=[REF(src)]'>here</a> to place your votes.\nYou have [DisplayTimeText(vp)] to vote.</font>"
+		if(observer_vote_allowed || ghost_vote_default_ok)
```
Comment (MarkSuckerberg) [inline code/controllers/subsystem/vote.dm]:
```suggestion
		if(observer_vote_allowed)
```
I think this is what you want

Comment (Maldaris) [reply code/controllers/subsystem/vote.dm]: yeah I just made this change 👍

**Comment by Maldaris** [comment]
> I haven't really paid attention to it, but does this effect observers being able to vote on map votes at the end of a round?

No, this only affects the ability for observers to vote on transfer.

**Pull request merged**